Unable to compile on MacOS

### Describe the bug

Both the automatic release process and myself compiling locally fails with the following error:
```
src/third-party/imgui/imgui_freetype.cpp:38:10: fatal error: 'ft2build.h' file not found
#include <ft2build.h>
         ^~~~~~~~~~~~
1 error generated.
```
which clearly indicates that libraries are not included and linked properly.

### Attach save file

N/A

### Steps to reproduce

On MacOS, run e.g. `make RELEASE=1 TILES=1 SOUND=1 LOCALIZE=0 CLANG=0` and watch the compilation fail.

### Expected behavior

The compilation succeeds.

### Screenshots

_No response_

### Versions and configuration

OS: MacOS Sequoia 15.0

### Additional context

The following addition to `Makefile` made the compilation succeed:
```
@@ -765,6 +765,8 @@ ifeq ($(TILES), 1)
     LDFLAGS += $(shell $(PKG_CONFIG) --libs freetype2)
   endif
 
+  CXXFLAGS += $(shell $(PKG_CONFIG) --cflags freetype2)
+  LDFLAGS += $(shell $(PKG_CONFIG) --libs freetype2)
   DEFINES += -DTILES
 
   ifeq ($(TARGETSYSTEM),WINDOWS)
```
So the problem is likely that the `freetype2` library is not included properly for MacOS. It should probably be quite easy to fix for someone more familiar with the makefile than me.

--- a/Makefile
+++ b/Makefile
@@ -750,6 +750,8 @@ ifeq ($(TILES), 1)
         LDFLAGS += -lSDL2_mixer
       endif
     endif
+    CXXFLAGS += $(shell $(PKG_CONFIG) --cflags freetype2)
+    LDFLAGS += $(shell $(PKG_CONFIG) --libs freetype2)
   else ifneq ($(NATIVE),emscripten)
     CXXFLAGS += $(shell $(PKG_CONFIG) --cflags sdl2)
     CXXFLAGS += $(shell $(PKG_CONFIG) --cflags SDL2_image SDL2_ttf)

